cg/meta/workflow/mutant/mutant.py

```diff
@@ -244,3 +209,81 @@ def link_nanopore_fastq_for_sample(
         LOG.info(f"Concatenation in progress for sample {sample.internal_id}.")
         self.fastq_handler.concatenate(read_paths, concatenated_path)
         self.fastq_handler.remove_files(read_paths)
+
+    def run_analysis(self, case_id: str, dry_run: bool, config_artic: str = None) -> None:
+        if self.get_case_output_path(case_id=case_id).exists():
+            LOG.info("Found old output files, directory will be cleaned!")
+            shutil.rmtree(self.get_case_output_path(case_id=case_id), ignore_errors=True)
+
+        if config_artic:
+            self.process.run_command(
+                [
+                    "analyse",
+                    "sarscov2",
+                    "--config_case",
+                    self.get_case_config_path(case_id=case_id).as_posix(),
+                    "--config_artic",
+                    config_artic,
+                    "--outdir",
+                    self.get_case_output_path(case_id=case_id).as_posix(),
+                    self.get_case_fastq_dir(case_id=case_id).as_posix(),
+                ],
+                dry_run=dry_run,
+            )
+        else:
+            self.process.run_command(
+                [
+                    "analyse",
+                    "sarscov2",
+                    "--config_case",
+                    self.get_case_config_path(case_id=case_id).as_posix(),
+                    "--outdir",
+                    self.get_case_output_path(case_id=case_id).as_posix(),
+                    self.get_case_fastq_dir(case_id=case_id).as_posix(),
+                ],
+                dry_run=dry_run,
+            )
+
+    def get_cases_to_store(self) -> list[Case]:
+        """Return cases with completed analysis on Trailblazer, a deliverables file in the case directory,
+        and ready to be stored in Housekeeper."""
+        return [
+            case
+            for case in self.status_db.get_running_cases_in_workflow(workflow=self.workflow)
+            if Path(self.get_deliverables_file_path(case_id=case.internal_id)).exists()
+        ]
+
+    def run_qc_and_fail_analyses(self, dry_run: bool) -> None:
+        """Run qc check, report qc summaries on Trailblazer and fail analyses that fail QC."""
+        for case in self.get_cases_to_store():
+            qc_result: QualityResult = self.get_qc_result(case=case)
+            if not qc_result:
+                LOG.error(f"Could not run QC for case {case.internal_id}")
+                self.trailblazer_api.set_analysis_status(
+                    case_id=case.internal_id, status=AnalysisStatus.ERROR
+                )
+                raise CgError
```

Comment:
Can a more specific exception be raised here that is more descriptive of what went wrong? Maybe `MutantQCError` or something like that.